lib/debug/server.rb

```diff
@@ -164,6 +164,7 @@ def process
           }
         end
 
+        return unless line
```

Comment:
If the `nil` is expected after the Socket is closed, I think it'd be clearer if we dear with it around line 162. For example:

```rb
         if l = @sock.gets
           DEBUGGER__.info "UI_Server received: #{l.chomp}"
         else
           # socket is closed
           return
         end
```

Reply:
Now the lines are already complicated so I left it.